receiver/mysqlreceiver/metadata.yaml

```
@@ -279,3 +289,21 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [threads]
+  mysql.perf.events.statements:
```

```suggestion
  mysql.statement_event.count:
```

[MySQL documentation](https://dev.mysql.com/doc/refman/5.6/en/performance-schema-events-statements-current-table.html) uses the term "statement events" when describing the subject in a sentence. This sounds to me like we should use `statement_event` as a combined namespace, since they are not separate things. (An event does not have multiple statements. Rather, a statement is a type of event.)

In a recent PR, it was decided not to include `perf` in the namespace, so I think we should exclude it here as well to be consistent.

Also, we cannot have a metric name that is a namespace of another metric. This is a count, right? Typically, we'll use `.count` for that, so as not to overload the namespace.

---

Makes sense :+1:

I have an issue with `COUNT_STAR` row. It seems to be an overall `count` as I understand it correctly

```
 COUNT_STAR

The number of summarized events. This value includes all events, whether timed or nontimed. 
```

Not sure about label value for it :shrug: 

ref: https://dev.mysql.com/doc/refman/5.6/en/performance-schema-wait-summary-tables.html

---

WDYT about `all`?

---

> The number of summarized events. This value includes all events, whether timed or nontimed. 

Based on this, the proper thing to do according to the OTel data model would be to exclude the data point entirely. The reasoning is that it will distort the derived total when aggregated with the other data points. 

Imagine a [stacked chart](https://static.anychart.com/images/gallery/v8/area-charts-stacked-area-chart.png) that contains each category, and also the total. The purpose of stacking the data rather than plotting each series relative to zero is to show a derived total. By throwing in a separate total, we'd show double the actual when stacked. There are other ways in which the data may be aggregated, but the same concept applies.

If it's really important to have an explicit "total" value, we should have a completely separate metric for it. Typically this is only necessary when the sum of the other data points does _not_ accurately describe the total. In theory this could be due to events that do not fall into any of the categories, or events that are counted in more than one category. (I suspect this is not the case here though.)

---

Stacked chart is very good point. I think there is no need to create this separate metric, so I will remove it `star` entirely